packages/x-date-pickers/src/internals/hooks/usePicker/usePickerProvider.ts

```diff
@@ -44,3 +123,25 @@ export function usePickerProvider<TValue>(
     privateContextValue,
   };
 }
+
+export interface UsePickerProviderParameters<TValue>
+  extends Pick<PickerProviderProps, 'localeText'> {
+  props: UsePickerProps<TValue, any, any, any, any>;
+  pickerValueResponse: UsePickerValueResponse<TValue, FieldSection, any>;
+  valueManager: PickerValueManager<TValue, any>;
+  variant: PickerVariant;
+  views: readonly DateOrTimeViewWithMeridiem[];
+}
+
+export interface UsePickerProviderReturnValue extends Omit<PickerProviderProps, 'children'> {}
+
+/**
+ * Props used to create the private context.
+ * Those props are exposed on all the pickers.
+ */
+export interface UsePickerProviderProps extends FormProps {
+  /**
+   * Force rendering in particular orientation.
```

Comment:
It's the JSDoc we currently have in `usePickersLayoutProps` but it's probably not great :thinking: 
If you have a proposal I can apply it in a follow up (to avoid poluting this PR with all the doc gen)